connector/connection.go

```diff
@@ -327,15 +333,37 @@ func (c *MQTTConnection) onConnected(client mqtt.Client) {
 			return true
 		}
 
-		logFields["qos"] = sub.qos
-		logFields["topics"] = sub.topics
+		subReq := subFilters(sub.qos, sub.topics)
+		logFields["filters"] = subReq
 
 		c.logger.Info("Sending subscribe packet", logFields)
-		token := c.mqttClient.SubscribeMultiple(subFilters(sub.qos, sub.topics), sub.callback)
+		token := c.mqttClient.SubscribeMultiple(subReq, sub.callback)
+		if subToken, ok := token.(*mqtt.SubscribeToken); ok {
+			subTokens = append(subTokens, subToken)
+		}
+		return true
+	})
+
+	c.waitForSubscribeTokens(subTokens)
+
+	c.fireConnectionEvent(true, nil)
+}
+
+func (c *MQTTConnection) waitForSubscribeTokens(tokens []*mqtt.SubscribeToken) {
+	logFields := watermill.LogFields{
+		"mqtt_url": c.config.URL,
+		"clientid": c.clientID,
```

Comment:
Could you please revise the usage of clientid log field? It should be: client_id. Note that even client_d is used and has to be fixed.

Reply:
clientid is still used, for example in unsubscribe.